tmail-backend/jmap/extensions-cassandra/src/main/scala/com/linagora/tmail/james/jmap/settings/CassandraJmapSettingsDAO.scala

```diff
@@ -0,0 +1,124 @@
+  val STATE: CqlIdentifier = CqlIdentifier.fromCql("state")
+  val SETTINGS: CqlIdentifier = CqlIdentifier.fromCql("settings")
+
+  val MODULE: CassandraModule = CassandraModule.table(TABLE_NAME)
+    .comment("Hold user JMAP settings")
+    .statement(statement => _ => statement
+      .withPartitionKey(USER, TEXT)
+      .withColumn(STATE, UUID)
+      .withColumn(SETTINGS, mapOf(TEXT, TEXT)))
+    .build
+
+  val MAP_OF_STRING_CODEC: TypeCodec[ImmutableMap[String,String]] = CodecRegistry.DEFAULT.codecFor(frozenMapOf(TEXT, TEXT))
+}
+
+class CassandraJmapSettingsDAO @Inject()(session: CqlSession) {
+  import CassandraJmapSettingsTable._
+
+  private val executor: CassandraAsyncExecutor = new CassandraAsyncExecutor(session)
+
+  private val insert: PreparedStatement = session.prepare(insertInto(TABLE_NAME)
+    .value(USER, bindMarker(USER))
+    .value(STATE, bindMarker(STATE))
+    .value(SETTINGS, bindMarker(SETTINGS))
+    .build())
+
+  private val updateStatement: PreparedStatement = session.prepare(update(TABLE_NAME)
+    .setColumn(STATE, bindMarker(STATE))
+    .setColumn(SETTINGS, bindMarker(SETTINGS))
+    .where(column(USER).isEqualTo(bindMarker(USER)))
+    .build())
+
+  private val selectAll: PreparedStatement = session.prepare(selectFrom(TABLE_NAME)
+    .all()
+    .whereColumn(USER).isEqualTo(bindMarker(USER))
+    .build())
+
+  private val selectOne: PreparedStatement = session.prepare(selectFrom(TABLE_NAME)
+    .all()
+    .whereColumn(USER).isEqualTo(bindMarker(USER))
+    .build())
+
+  private val deleteOne: PreparedStatement = session.prepare(deleteFrom(TABLE_NAME)
+    .whereColumn(USER).isEqualTo(bindMarker(USER))
+    .build())
+
+  def insert(username: Username, jmapSettings: JmapSettings): SMono[JmapSettings] = {
+    val settings: Map[String, String] = jmapSettings.settings.map(item => (item._1.value.value, item._2.value))
+    val insertSetting = insert.bind()
+      .set(USER, username.asString(), TypeCodecs.TEXT)
+      .set(STATE, jmapSettings.state.value.toString, TypeCodecs.TEXT)
+      .set(SETTINGS, settings , MAP_OF_STRING_CODEC)
+
+    SMono.fromPublisher(executor.executeVoid(insertSetting)
+      .thenReturn(jmapSettings))
+  }
+
+  def selectAll(username: Username): SFlux[JmapSettings] =
+    SFlux.fromPublisher(executor.executeRows(selectAll.bind()
+      .set(USER, username.toString, TypeCodecs.TEXT))
+      .map(toJmapSettings))
+
+  def selectOne(username: Username): SMono[JmapSettings] =
+    SMono.fromPublisher(executor.executeSingleRow(selectOne.bind()
+      .set(USER, username.asString(), TypeCodecs.TEXT))
+      .map(toJmapSettings))
+
+  def selectState(username: Username): SMono[UuidState] =
+    SMono.fromPublisher(executor.executeSingleRow(selectOne.bind()
+      .set(USER, username.asString(), TypeCodecs.TEXT))
+      .map(toState))
+
+  def updateSetting(username: Username, newState: Option[UuidState], newSettings: Option[Map[JmapSettingsKey, JmapSettingsValue]]): SMono[Void] = {
```

Comment:
Option[UuidState] -> UuidState 
when we update setting, we always have a new state

Reply:
Same with `newSettings: Option[Map[JmapSettingsKey, JmapSettingsValue]]`